ocs_ci/ocs/dr_upgrade.py

```diff
@@ -0,0 +1,134 @@
+"""
+All DR operators upgrades implemented here ex: MulticlusterOrchestrator, Openshift DR operator
+
+"""
+
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.ocs.cluster import CephCluster, CephClusterExternal, CephHealthMonitor
+from ocs_ci.ocs.exceptions import TimeoutException
+from ocs_ci.ocs.ocs_upgrade import OCSUpgrade, verify_image_versions
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import defaults
+from ocs_ci.deployment.helpers.external_cluster_helpers import (
+    ExternalCluster,
+    get_external_cluster_client,
+)
+from ocs_ci.ocs.resources.install_plan import wait_for_install_plan_and_approve
+from ocs_ci.utility.utils import TimeoutSampler
+
+
+log = logging.getLogger(__name__)
+
+DR_TO_CEPH_CLUSTER_MAP = {"regional-dr": CephCluster, "metro-dr": CephClusterExternal}
+
+
+class DRUpgrade(OCSUpgrade):
+    """
+    Base class for all DR operator upgrades
+
+    """
+
+    def __init__(
+        self,
+        namespace=constants.OPENSHIFT_OPERATORS,
+        version_before_upgrade=config.ENV_DATA.get("ocs_version"),
+        ocs_registry_image=config.UPGRADE.get("upgrade_ocs_registry_image"),
+        upgrade_in_current_source=config.UPGRADE.get(
+            "upgrade_in_current_source", False
+        ),
+    ):
+        self.namespace = (
+            namespace if namespace else config.ENV_DATA["cluster_namespace"]
+        )
+        self.version_before_upgrade = version_before_upgrade
+        self.ocs_registry_image = ocs_registry_image
+        self.upgrade_in_current_source = upgrade_in_current_source
+        self.ceph_cluster = DR_TO_CEPH_CLUSTER_MAP[
+            config.MULTICLUSTER["multicluster_mode"]
+        ]
+        self.external_cluster = None
+        self.operator_name = None
+
+        super.__init__(
+            self.namespace,
+            self.version_before_upgrade,
+            self.ocs_registry_image,
+            self.upgrade_in_current_source,
+        )
+
+    def run_upgrade(self):
+        self.upgrade_version = self.get_upgrade_version()
+        assert self.get_parsed_versions()[1] >= self.get_parsed_versions()[0], (
+            f"Version you would like to upgrade to: {self.upgrade_version} "
+            f"is not higher or equal to the version you currently running: "
+            f"{self.version_before_upgrade}"
+        )
+
+        # create external cluster object
+        if config.DEPLOYMENT["external_mode"]:
+            host, user, password, ssh_key = get_external_cluster_client()
+            self.external_cluster = ExternalCluster(host, user, password, ssh_key)
+        self.csv_name_pre_upgrade = self.get_csv_name_pre_upgrade()
+        self.pre_upgrade_images = self.get_pre_upgrade_image(self.csv_name_pre_upgrade)
+        self.load_version_config_file(self.upgrade_version)
+
+        with CephHealthMonitor(self.ceph_cluster):
```

Comment:
Need to instantiate self.ceph_cluster

Reply:
Suggest dropping out the ceph health monitor for now, as this will require more adjustment in the ceph cluster object and also in the health monitor context manager to enforce it to run on specific cluster and propagate specific kubeconfig for all of underlaying calls.